openfisca_core/entities.py

```diff
@@ -51,9 +53,23 @@ def check_role_validity(self, role):
         if role is not None and not type(role) == Role:
             raise Exception("{} is not a valid role".format(role))
 
-    def __call__(self, variable_name, period, options = [], **parameters):
+    def check_period_validity(self, variable_name, period):
+        if period is None:
+            stack = traceback.extract_stack()
+            filename, line_number, function_name, line_of_code = stack[-3]
+            raise ValueError(u'''
```

Comment:
In Python 2 exception message must be encoded (in UTF-8)

`raise ValueError(u'xxx {}'.format(yyy).encode('utf-8'))`

(If one of the variables mentioned in `.format` contains unicode characters, Python 2 won't take the liberty to choose UTF-8 encoding, and this will raise a `UnicodeEncodeError` which has nothing to do with the original `ValueError`)